yjit/src/backend/x86_64/mod.rs

```diff
@@ -354,7 +354,7 @@ impl Assembler
                 },
                 Opnd::UImm(value) => {
                     // 32-bit values will be sign-extended
-                    if imm_num_bits(*value as i64) > 32 {
+                    if uimm_num_bits(*value as u64) > 32 {
```

Comment:
This is intentional because the value will get sign extended. We want to preserve the unsigned value passed in post sign extension we need to treat it as a signed integer.